typescript/sdk/src/deploy/verify/ContractVerifier.ts

```diff
@@ -7,243 +7,237 @@ import { sleep, strip0x } from '@hyperlane-xyz/utils';
-      this.logger(
-        `[${chain}] [${input.name}] Error checking if contract is already verified: ${error}`,
+      verificationLogger(
+        `Error checking if contract is already verified: ${error}`,
       );
       return false;
     }
   }
 
-  async verifyProxy(
+  private async verifyProxy(
     chain: ChainName,
     input: ContractVerificationInput,
+    verificationLogger: Debugger,
   ): Promise<void> {
-    if (input.isProxy) {
-      try {
-        const proxyGuid = await this.submitForm(
-          chain,
-          ExplorerApiActions.MARK_PROXY,
-          {
-            address: input.address,
-          },
-        );
-
-        const addressUrl = await this.multiProvider.tryGetExplorerAddressUrl(
-          chain,
-          input.address,
-        );
-
-        // poll for verified proxy status
-        if (proxyGuid) {
-          await this.submitForm(chain, ExplorerApiActions.CHECK_PROXY_STATUS, {
-            guid: proxyGuid,
-          });
-          this.logger(
-            `[${chain}] [${input.name}] Successfully verified proxy ${addressUrl}#readProxyContract`,
-          );
-        }
-      } catch (error) {
-        console.error(
-          `[${chain}] [${input.name}] Verification of proxy at ${input.address} failed`,
-        );
-        throw error;
-      }
+    if (!input.isProxy) return;
+
+    try {
+      const proxyGuid = await this.submitForm(
+        chain,
+        ExplorerApiActions.MARK_PROXY,
+        verificationLogger,
+        { address: input.address },
+      );
+      if (!proxyGuid) return;
+
+      await this.submitForm(
+        chain,
+        ExplorerApiActions.CHECK_PROXY_STATUS,
+        verificationLogger,
+        {
+          guid: proxyGuid,
+        },
+      );
+      const addressUrl = await this.multiProvider.tryGetExplorerAddressUrl(
+        chain,
+        input.address,
+      );
+      verificationLogger(
+        `Successfully verified proxy ${addressUrl}#readProxyContract`,
+      );
+    } catch (error) {
+      verificationLogger(
+        `Verification of proxy at ${input.address} failed: ${error}`,
+      );
+      throw error;
     }
   }
 
-  async verifyImplementation(
+  private async verifyImplementation(
     chain: ChainName,
     input: ContractVerificationInput,
+    verificationLogger: Debugger,
   ): Promise<void> {
-    this.logger(
-      `[${chain}] [${input.name}] Verifying implementation at ${input.address}`,
-    );
+    verificationLogger(`Verifying implementation at ${input.address}`);
+
+    const sourceName = this.contractMap[input.name];
+    if (!sourceName) {
+      const errorMessage = `Contract '${input.name}' not found in provided build artifact`;
+      verificationLogger(errorMessage);
+      throw new Error(`[${chain}] ${errorMessage}`);
+    }
 
     const data = {
-      sourceCode: this.source,
-      contractname: input.name,
+      sourceCode: this.standardInputJson,
+      contractname: `${sourceName}:${input.name}`,
```

Comment:
interesting
do we need the name on the input separately?

Reply:
yup the expected input on the explorer end is a fully qualified name, something like  `solidity/tokens/Tokens.sol:HypERC20`

this is very much an etherscan specific thing, blockscout don't even require you to provide a name